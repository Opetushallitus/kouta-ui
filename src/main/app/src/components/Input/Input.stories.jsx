import React from 'react';
import { storiesOf } from '@storybook/react';

import Input from './index';
import InputIcon from '../InputIcon';

storiesOf('Input', module)
  .add('Basic', () => <Input />)
  .add('With disabled', () => <Input disabled />)
  .add('With error', () => <Input error />)
  .add('With addon', () => (
    <Input suffix={<InputIcon type="access_time" />} />
  ));
