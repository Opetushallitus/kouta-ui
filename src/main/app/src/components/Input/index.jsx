export { default } from '@opetushallitus/virkailija-ui-components/Input';
