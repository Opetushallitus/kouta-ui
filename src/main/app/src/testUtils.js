import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
