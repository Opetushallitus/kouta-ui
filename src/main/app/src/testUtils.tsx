import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
