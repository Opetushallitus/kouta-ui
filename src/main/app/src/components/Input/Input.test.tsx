import React from 'react';

import Input from './index';
import { mountWithTheme } from '../../testUtils';

test('renders correctly', () => {
  expect(mountWithTheme(<Input />)).toMatchSnapshot();
});
