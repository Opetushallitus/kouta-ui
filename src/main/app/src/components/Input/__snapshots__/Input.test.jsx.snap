// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  border: 1px solid #ced4da;
  color: #666666;
  font-family: 'Roboto',sans-serif;
  line-height: 1.5;
  border-radius: 4px;
  outline: none;
  padding: 6px 12px;
  font-size: 1rem;
  -webkit-transition: border-color 0.25s,box-shadow 0.25s;
  transition: border-color 0.25s,box-shadow 0.25s;
  display: block;
  width: 100%;
  box-sizing: border-box;
  background-color: white;
  box-shadow: 0 0 0 0 transparent;
}

.c1:hover,
.c1:focus {
  border-color: #159ecb;
}

.c1:focus {
  box-shadow: 0 0 0 3px rgba(21,158,203,0.3);
}

.c0 {
  position: relative;
}

<ForwardRef>
  <styled.div>
    <div
      className="c0"
    >
      <styled.input
        disabled={false}
        error={false}
        hasPrefix={false}
        hasSuffix={false}
      >
        <input
          className="c1"
          disabled={false}
          type="text"
        />
      </styled.input>
    </div>
  </styled.div>
</ForwardRef>
`;
